tests/test_unknown_platforms.py

```diff
@@ -72,3 +88,174 @@ def test_get_sensor_given_type(self):
         assert sensor.sensor_type_name == "Radar"
         assert sensor.host__name == "Test Name"
         assert sensor.privacy_name == "Public"
+
+
+def test_unknown_platform_from_importer():
+    class TestParser(Importer):
+        def __init__(
+            self,
+            name="Test Importer",
+            validation_level=constants.NONE_LEVEL,
+            short_name="Test Importer",
+            datafile_type="Test",
+        ):
+            super().__init__(name, validation_level, short_name, datafile_type)
+            self.errors = list()
+
+        def can_load_this_header(self, header) -> bool:
+            return True
+
+        def can_load_this_filename(self, filename):
+            return True
+
+        def can_load_this_type(self, suffix):
+            return True
+
+        def can_load_this_file(self, file_contents):
+            return True
+
+        def _load_this_file(self, data_store, path, file_object, datafile, change_id):
+            self.test_platform = self.get_cached_platform(
+                data_store, None, change_id=change_id, unknown=True
+            )
+
+    ds = DataStore("", "", "", 0, ":memory:", db_type="sqlite")
+    ds.initialise()
+
+    with ds.session_scope():
+        ds.populate_reference()
+        ds.populate_metadata()
+
+    test_parser = TestParser()
+
+    processor = FileProcessor()
+    processor.register_importer(test_parser)
+
+    with ds.session_scope():
+        processor.process(SINGLE_REP_FILE, ds, False)
+        ds.session.add(test_parser.test_platform)
+        _ = uuid.UUID(test_parser.test_platform.name)
+        assert test_parser.test_platform.platform_type_name == "Unknown"
+        assert test_parser.test_platform.nationality_name == "Unknown"
+
+
+class UnknownPlatformsResolverTestCase(unittest.TestCase):
+    def setUp(self):
+        self.ds = DataStore(
+            "", "", "", 0, ":memory:", db_type="sqlite", missing_data_resolver=CommandLineResolver()
+        )
+        self.ds.initialise()
+
+        with self.ds.session_scope():
+            self.ds.populate_reference()
+
+        self.processor = FileProcessor()
+        self.processor.load_importers_dynamically()
+
+    @patch("pepys_import.file.file_processor.prompt")
+    @patch("pepys_import.resolvers.command_line_input.prompt")
+    @patch("pepys_import.resolvers.command_line_resolver.prompt")
+    def test_answering_store_as_unknown(self, prompt, menu_prompt, processor_prompt):
```

Comment:
@robintw - the linter is saying `prompt` is unused.  It is unused, but is it still a necessary import?